cli/cdsctl/template.go

```diff
@@ -105,43 +107,79 @@ var templatePushCmd = cli.Command{
 func templatePushRun(v cli.Values) error {
 	files := strings.Split(v.GetString("yaml-file"), ",")
 
-	buf := new(bytes.Buffer)
+	// create a new tar archive
 	var dir string
+	tar := new(bytes.Buffer)
 
-	// Create a new tar archive.
-	filesToRead := []string{}
-	for _, file := range files {
-		fi, err := os.Lstat(file)
+	// if the first args is an url, try to download all files
+	readFromLink := len(files) > 0 && exportentities.IsURL(files[0]) && strings.HasSuffix(files[0], ".yml")
+	if readFromLink {
+		manifestURL := files[0]
+		baseURL := manifestURL[0:strings.LastIndex(manifestURL, "/")]
+
+		// get the manifest file
+		contentFile, _, err := exportentities.OpenPath(manifestURL)
 		if err != nil {
-			fmt.Printf("Skipping file %s: %v\n", file, err)
-			continue
+			return err
+		}
+		buf := new(bytes.Buffer)
+		if _, err := buf.ReadFrom(contentFile); err != nil {
+			return fmt.Errorf("Cannot read from given remote file: %v", err)
+		}
+		var t exportentities.Template
+		if err := yaml.Unmarshal(buf.Bytes(), &t); err != nil {
+			return fmt.Errorf("Cannot unmarshal given remote yaml file: %v", err)
 		}
 
-		//Skip the directory
-		if fi.IsDir() {
-			continue
+		// get all components of the template
+		paths := []string{t.Workflow}
+		paths = append(paths, t.Pipelines...)
+		paths = append(paths, t.Applications...)
+		paths = append(paths, t.Environments...)
+
+		links := make([]string, len(paths))
+		for i := range paths {
+			links[i] = fmt.Sprintf("%s/%s", baseURL, paths[i])
 		}
 
-		fmt.Println("Reading file ", cli.Magenta(file))
-		if dir == "" {
-			dir = filepath.Dir(file)
+		if err := workflowLinksToTarWriter(append(links, manifestURL), tar); err != nil {
+			return err
 		}
-		if dir != filepath.Dir(file) {
-			return fmt.Errorf("files must be ine the same directory")
+	} else {
+		filesToRead := []string{}
+		for _, file := range files {
+			fi, err := os.Lstat(file)
+			if err != nil {
+				fmt.Printf("Skipping file %s: %v\n", file, err)
```

Comment:
Error strings should not start with a capital letter because they'll often be prefixed before printing:

https://github.com/golang/go/wiki/Errors